File: exercises.js
// Easy
// 1. Reverse Integer (JS)

var reverse = function(x) {
    const n = parseInt(String(Math.abs(x)).toString().split('').reverse().join(''))
    return n > Math.pow(2,31) - 1 ? 0 : x > 0 ? n : n * - 1  
}

// 2. Two Sum 

var twoSum = function(nums, target) {
    const result = {}
    for (let i = 0; i < nums.length; i++) {
        if (result[nums[i]] >= 0){
            return [result[nums[i]], i]
        } 
        result[target-nums[i]] = i
    }
};

// 3. Shuffle String

var restoreString = function(s, indices) {
    var len = indices.length
    var str = new Array(len)
    for(let i = 0; i < len; i++){
        str[indices[i]] = s[i]
        }
    return str.join("")
};

// ------------------------------------------------

// Medium

// ------------------------------------------------

// Hard 

File: exercises.py
# Easy
# 1. Palindrome Number

# class Solution(object):
#     def isPalindrome(self, x):
#         return False if x < 0 else x == int(str(x)[::-1])
    
# ------------------------------------------------

# Medium

class Solution(object):
    def findDuplicate(self, paths):
        """
        :type paths: List[str]
        :rtype: List[List[str]]
        """
        if not paths:
            return []
        
        dict_ = defaultdict(list)
        
        for path in paths:
            xPath = path.split()
            
            for x in xPath[1:]:
                dict_[x[x.find('(') +1 : -1]].append('/'.join([xPath[0], x[:x.find("(")]]))
                
        return [i for i in dict_.values() if len(i) > 1]

# ------------------------------------------------

# Hard 

